src/components/combo_box/combo_box.tsx

```diff
@@ -936,6 +936,10 @@ export class EuiComboBox<T> extends Component<
       matchingOptions,
     } = this.state;
 
+    // Make sure we have a valid ID if users don't pass one as a prop
+    const inputId =
+      id === undefined || id === null ? this.rootId('_eui-combobox-id') : id;
```

Comment:
I'd probably opt to go with a simple falsey check on this conditional and catch edge case empty strings as well
```suggestion
    const inputId = id || this.rootId('_eui-combobox-id');
```

if for some reason empty strings are valid/accessible, you could do a non strict equal check to capture both undefined and null

```suggestion
    const inputId = id !=  null ? this.rootId('_eui-combobox-id') : id;
```

Reply:
Go with the second one. `0` is a valid id that would mistakenly be capture by the first.

Reply:
General question here. While `0` is **technically** a valid `id`, do we really care to support it as one? I'm just thinking about an eng who decides to use simply the `index` of an item as the `id` which if done multiple times to different components would result in multiple of the same `id`s.

Reply:
Was just about to basically post what Caroline said, haha. How often are we really expecting folks to pass in numbers vs strings for IDs? 🤷 If we're really concerned, can we fall back to `??` instead as someone suggested in Slack?

```suggestion
    const inputId = id ?? this.rootId('_eui-combobox-id');
```

Reply:
I think I overcomplicated this discussion. `id` can't be `0` because it's required to be a string, which also negates the discussion about what we want to allow consumers to provide.
So `||` or `??` are both acceptable here.